### PR TITLE
fix: Docs Site - Table of contents highlight not working

### LINKED ISSUE #4658: Docs Site - Table of contents highlight not working
### What version of `astro` are you using?

1.1.3

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

We started using Astro for our [Media Chrome docs (PR)](https://github.com/muxinc/media-chrome/pull/292) and really like it so far!

One thing we noticed is that the table of contents doesn't seem to highlight the current title of the section you're in when you scroll. Is this expected? 

Thanks!

### Link to Minimal Reproducible Example

https://media-chrome-docs-git-fork-luwes-docs-site-mux.vercel.app/en/get-started

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fixes #4658

### CODE REVIEW

**Comment by rishi-raj-jain** [comment]
@matthewp @sarah11918

**Review by matthewp — APPROVED** [review]
great, thank you!

File: examples/docs/src/components/RightSidebar/TableOfContents.tsx [hunk 1]
```diff
@@ -30,22 +32,57 @@ const TableOfContents: FunctionalComponent<{ headings: MarkdownHeading[] }> = ({
...
 						<li
-							className={`heading-link depth-${heading.depth} ${
-								activeId === heading.slug ? 'active' : ''
+							className={`header-link depth-${heading.depth} ${
+								currentID === heading.slug ? 'current-header-link' : ''
 							}`.trim()}
 						>
-							<a href={`#${heading.slug}`}>{heading.text}</a>
+							<a href={`#${heading.slug}`} onClick={onLinkClick}>
+								{unescape(heading.text)}
```
Comment (matthewp) [inline examples/docs/src/components/RightSidebar/TableOfContents.tsx:84]: do we need this? It was added before. Seems unrelated to the bug.

**Review by matthewp — CHANGES_REQUESTED** [review]
The dependency that was added is not part of the package.json, which is why CI is failing. Is it really needed though?  I'd like to see this removed if not critical. But if it is, make sure its in the example's package.json deps.

File: examples/docs/src/components/RightSidebar/TableOfContents.tsx (hunk 1 above)
Comment (rishi-raj-jain) [reply examples/docs/src/components/RightSidebar/TableOfContents.tsx:84]: @matthewp It's another good to have. Picked this up from https://github.com/withastro/docs/blob/main/src/components/RightSidebar/TableOfContents.tsx#L1 & https://github.com/withastro/docs/blob/main/src/components/RightSidebar/TableOfContents.tsx#L115

**Comment by rishi-raj-jain** [comment]
> The dependency that was added is not part of the package.json, which is why CI is failing. Is it really needed though? I'd like to see this removed if not critical. But if it is, make sure its in the example's package.json deps.

I mean it makes sense to have it there, making the docs example one closer to what docs.astro.build is where right now.
Better handling of text by escaping is a nice-to-have (as like this PR haha).

**Review by matthewp — APPROVED** [review]

**Comment by luwes** [comment]
thank you @rishi-raj-jain 👍

**Comment by rishi-raj-jain** [comment]
> thank you @rishi-raj-jain 👍

Pleasure!

**Pull request merged**